src/main/scala/org/broadinstitute/dsde/firecloud/dataaccess/ElasticSearchDAOQuerySupport.scala

```diff
@@ -0,0 +1,147 @@
+package org.broadinstitute.dsde.firecloud.dataaccess
+
+import org.broadinstitute.dsde.firecloud.model._
+import org.broadinstitute.dsde.firecloud.model.ModelJsonProtocol._
+import org.elasticsearch.search.aggregations.{AggregationBuilders, Aggregations}
+import org.elasticsearch.client.transport.TransportClient
+import org.elasticsearch.action.search.{SearchRequest, SearchRequestBuilder, SearchResponse}
+import org.elasticsearch.index.query.{BoolQueryBuilder, QueryBuilder}
+import org.elasticsearch.index.query.QueryBuilders._
+import org.elasticsearch.search.aggregations.bucket.terms.Terms
+import spray.json._
+import spray.json.DefaultJsonProtocol._
+
+import scala.collection.JavaConverters._
+import scala.concurrent.Future
+import scala.concurrent.ExecutionContext.Implicits.global
+
+
+trait ElasticSearchDAOQuerySupport extends ElasticSearchDAOSupport {
+
+  /** ES queries - below is similar to what will be created by the query builders
+    * {"query":{"match_all":{}}}"
+    * {"query":{
+    *    "bool":{
+    *      "must":[{
+    *        "bool":{
+    *        "should":[
+    *          {"term":{"library:indication":"n/a"}},
+    *          {"term":{"library:indication":"disease"}},
+    *          {"term":{"library:indication":"lukemia"}}]
+    *        }
+    *      },
+    *      {"match":{"all_":"broad"}}]}},
+    *  "aggregations":{
+    *    "library:dataUseRestriction":{
+    *      "terms":{"field":"library:dataUseRestriction.raw"}},
+    *    "library:datatype":{
+    *      "terms":{"field":"library:datatype.raw"}}
+    * }
+    *  The outer boolean query, which is a must, is indicating that all selections of the filter and search
+    *  are being "and"-ed together
+    *  For selections with an attribute, each selection is "or"-ed together - this is represented by the inner
+    *  boolean query with the should list
+    */
+
+
+  def createQuery(criteria: LibrarySearchParams): QueryBuilder = {
+    val query: BoolQueryBuilder = boolQuery // outer query, all subqueries should be added to the must list
+    query.must(criteria.searchString match {
+      case None => matchAllQuery
+      case Some(searchTerm) if searchTerm.trim == "" => matchAllQuery
+      case Some(searchTerm) => matchQuery("_all", searchTerm)
+    })
+    criteria.searchFields foreach { case (field:String, values:Seq[String]) =>
+      val fieldQuery = boolQuery // query for possible values of aggregation, added via should
+      values foreach { value:String => fieldQuery.should(termQuery(field+".raw", value))}
+      query.must(fieldQuery)
+    }
```

Comment:
I think - after digging into the query construction a bit - that this is more correct:
```
    criteria.searchFields foreach { case (field:String, values:Seq[String]) =>
      values foreach { value:String => query.filter(termQuery(field+".raw", value))}
    }
```
it does two things:
* remove the nested `must(should(term())`
* uses filter context for facet values, instead of the query context used for fulltext search (see https://www.elastic.co/guide/en/elasticsearch/reference/2.4/query-filter-context.html)